src/ace/managers/viewport/simplebuffer.c

```diff
@@ -90,151 +200,98 @@ tSimpleBufferManager *simpleBufferCreate(
-		uwModulo -= 1;
-	}
-	logWrite("Modulo: %u\n", uwModulo);
-
-	// Update (rewrite) copperlist
-	// TODO this could be unified with copBlock being set with copSetMove too
-	tCopList *pCopList = pManager->sCommon.pVPort->pView->pCopList;
-	if(pManager->ubFlags & SIMPLEBUFFER_FLAG_COPLIST_RAW) {
-		// Since simpleBufferProcess only updates bitplane ptrs and shift,
-		// copperlist must be shaped here.
-		// WAIT is calc'd in same way as in copBlockCreate in simpleBufferCreate().
-		tCopCmd *pCmdList = &pCopList->pBackBfr->pList[pManager->uwCopperOffset];
-		logWrite("Setting copperlist %p at offs %u: %p\n", pCopList->pBackBfr, pManager->uwCopperOffset, pCmdList);
-		copSetWait(&pCmdList[0].sWait, 0xE2-7*4, pManager->sCommon.pVPort->uwOffsY + 0x2C-1);
-		copSetMove(&pCmdList[1].sMove, &g_pCustom->ddfstop, 0x00D0);    // Data fetch
-		copSetMove(&pCmdList[2].sMove, &g_pCustom->ddfstrt, uwDDfStrt);
-		copSetMove(&pCmdList[3].sMove, &g_pCustom->bpl1mod, uwModulo);  // Bitplane modulo
-		copSetMove(&pCmdList[4].sMove, &g_pCustom->bpl2mod, uwModulo);
-		copSetMove(&pCmdList[5].sMove, &g_pCustom->bplcon1, 0);         // Shift: 0
-		UBYTE i;
-		ULONG ulPlaneAddr;
-		for (i = 0; i != pManager->sCommon.pVPort->ubBPP; ++i) {
-			ulPlaneAddr = (ULONG)pManager->pBuffer->Planes[i];
-			copSetMove(&pCmdList[6 + i*2 + 0].sMove, &g_pBplFetch[i].uwHi, ulPlaneAddr >> 16);
-			copSetMove(&pCmdList[6 + i*2 + 1].sMove, &g_pBplFetch[i].uwLo, ulPlaneAddr & 0xFFFF);
-		}
-		// Copy to front buffer since it needs initialization there too
-		CopyMem(
-			&pCopList->pBackBfr->pList[pManager->uwCopperOffset],
-			&pCopList->pFrontBfr->pList[pManager->uwCopperOffset],
-			(6+2*pManager->sCommon.pVPort->ubBPP)*sizeof(tCopCmd)
-		);
-	}
-	else {
-		tCopBlock *pBlock = pManager->pCopBlock;
-		pBlock->uwCurrCount = 0; // Rewind to beginning
-		copMove(pCopList, pBlock, &g_pCustom->ddfstop, 0x00D0);     // Data fetch
-		copMove(pCopList, pBlock, &g_pCustom->ddfstrt, uwDDfStrt);
-		copMove(pCopList, pBlock, &g_pCustom->bpl1mod, uwModulo);   // Bitplane modulo
-		copMove(pCopList, pBlock, &g_pCustom->bpl2mod, uwModulo);
-		copMove(pCopList, pBlock, &g_pCustom->bplcon1, 0);          // Shift: 0
-		UBYTE i;
-		ULONG ulPlaneAddr;
-		for (i = 0; i != pManager->sCommon.pVPort->ubBPP; ++i) {
-			ulPlaneAddr = (ULONG)pManager->pBuffer->Planes[i];
-			copMove(pCopList, pBlock, &g_pBplFetch[i].uwHi, ulPlaneAddr >> 16);
-			copMove(pCopList, pBlock, &g_pBplFetch[i].uwLo, ulPlaneAddr & 0xFFFF);
+	if(pManager) {
+		if(pManager->pCamera && isCameraCreated) {
+			cameraDestroy(pManager->pCamera);
 		}
+		memFree(pManager, sizeof(tSimpleBufferManager));
 	}
-	logBlockEnd("simplebufferSetBitmap()");
+	logBlockEnd("simpleBufferCreate()");
+	va_end(vaTags);
+	return 0;
 }
 
 void simpleBufferDestroy(tSimpleBufferManager *pManager) {
 	logBlockBegin("simpleBufferDestroy()");
 	logWrite("Destroying bitmap...\n");
-	bitmapDestroy(pManager->pBuffer);
+	if(pManager->pBack != pManager->pFront) {
```

Comment:
Can `pBack` be zero too?

Reply:
shouldn't be - in `simpleBufferCreate()` `pFront` gets created and if dbl bfr then `pBack` gets allocated, otherwise it's set to `pFront`. So freeing `pBack` should only take place when there's dlb buffering. Unless users is an idiot which pokes with code too much.

On the second thought I wonder if I should've added safety in scroll buffer.